### PR TITLE
pnpm 4.0.2 (new formula)

### PR DESCRIPTION
**Pull request by JounQin**
- [x] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/master/CONTRIBUTING.md)?
- [x] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [x] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [x] Does your build pass `brew audit --strict <formula>` (after doing `brew install <formula>`)?

-----

### CODE REVIEW

**Comment by SMillerDev** [comment]
Could you move the require inside the class please?

**Comment by JounQin** [comment]
Sorry I'm not so familiar with ruby, maybe a stupid question, what's the difference?

And I remember it's outside in `now-cli` formula.

**Comment by SMillerDev** [comment]
It differs in the moment it loads the dependency. Either on class usage or when the file is opened.

**Comment by JounQin** [comment]
@SMillerDev Done.

File: Formula/pnpm.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
+class Pnpm < Formula
+  desc "📦🚀 Fast, disk space efficient package manager "
```
Comment (alebcay) [inline Formula/pnpm.rb]:
```
Error: 1 problem in 1 formula detected
pnpm:
  * C: 2: col 54: Description shouldn't have a trailing space
```

Remove that trailing space and we should be good to go. 🎉

Comment (JounQin) [reply Formula/pnpm.rb]: Didn't notice that, thx.

**Comment by alebcay** [comment]
@BrewTestBot test this please.

File: Formula/pnpm.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
+class Pnpm < Formula
+  desc "📦🚀 Fast, disk space efficient package manager"
+  homepage "https://pnpm.js.org"
+  url "https://registry.npmjs.org/pnpm/-/pnpm-4.0.2.tgz"
+  sha256 "6f9b5faa2db1c2957e759c853e1e82bfb81fd7d7183858b8e794fb446640a988"
+
+  depends_on "node"
+
+  def install
+    require "language/node"
```
Comment (SMillerDev) [inline Formula/pnpm.rb]: Please move this to the start of the class

Comment (JounQin) [reply Formula/pnpm.rb]: @SMillerDev Please review again.

**Comment by JounQin** [comment]
@alebcay @SMillerDev 
Is there anything blocking then?

**Review by chenrui333 — APPROVED** [review]
lgtm

**Comment by chenrui333** [comment]
Thanks @JounQin for your contribution to Homebrew! :tada: 🥇 

Without awesome contributors like you, it would be impossible to maintain Homebrew to the high level of quality users have come to expect. Thank you!!!!

**Pull request closed without merging**